### ISSUE TITLE
Boat Location Indicator is missing from the Route as you step through the grib file.

### ISSUE THREAD

**Issue by rgleason** [seandepagnier/weather_routing_pi#123]
As a result of this Tracker request, I tried doing both multiple cmputations at the same time and stepping through a grib file with both computations highlighted at the same time. The result is that it works but now the Boat Indicator in missing!!

Also in order to find out where the boat indicator should be located on the route, the data at the cursor is not useful without having Data and Time information!

![screenshot 348](https://user-images.githubusercontent.com/2363959/35237083-88bc48d0-ff77-11e7-827d-0aa001db8874.png)
Showing calculation and showing of two routes at the same time. Boat is shown at the head.

![screenshot 349](https://user-images.githubusercontent.com/2363959/35237092-8d7cd2d6-ff77-11e7-9269-d32d35f85639.png)
Calc is complete, then try to step through the grib file, and there is no Boat Location shown!!

**Comment by rgleason** [comment]
I just checked and Weather_Routing 1.10 also has this problem. The Boat Position does not work!

**Comment by rgleason** [comment]
The Boat refuses to move out of the Destination Harbor and is completely unresponsive to the time of the Grib file, such that trying to step through the Grib file and changing grib time, does not seem to affect or change the Boat time or position. 

Somehow the grib time clock no longer controls the movement of the boat, after the computation is completed.

![screenshot 352](https://user-images.githubusercontent.com/2363959/35254804-dc067d98-ffb9-11e7-82f0-ab45278aa2ff.png)

**Comment by stelian42** [comment]
This works ok for me (under Linux).

I have used the default configuration, but downloaded a worldwide gfs grib (http://grib.virtual-loup-de-mer.org/latest.grb), and edited each configuration to use Grib (instead of climatology) and changed the start date to the date of the grib). Then compute.

When I change the grib timeline in the grib window, I can very well see the circle on the route (and it works on all the (selected) routes simultaneously).

![multiple-routing-boat-position](https://user-images.githubusercontent.com/548364/35266694-4df8bb04-0024-11e8-8af3-378dcc95ec7a.png)

**Comment by stelian42** [comment]
Just tried under Windows (inside a VirtualBox) and the indicator is there too...

![routing-virtualbox](https://user-images.githubusercontent.com/548364/35271094-4dccf91c-0031-11e8-9958-cf54c5eacf62.png)

**Comment by rgleason** [comment]
Thank you Stelian, could it be my compile?  I am using TDan's Batchutil  and "build clean" then "build package"  I'll try a fresh git clone.

**Comment by rgleason** [comment]
Which version of opencpn.lib is needed?   I was using one with 138kb from 1/4/2018
which must be api 1.15

I am going to try the 4.8.0 version which is 135kb.

**Comment by stelian42** [comment]
I used an older (december ?) version of opencpn.lib. I also tried with the latest opencpn.lib (the one obtained by building the latest git version of opencpn). Works fine too.

You cant find opencpn.lib here: http://popies.net/tmp/opencpn.lib , and the WR installer here: http://popies.net/tmp/weather_routing_pi-1.11-win32-sp20180123.exe . Give them a try.

**Comment by rgleason** [comment]
Ok, that might have been the problem, but now I realize that the boat only appears only 3 times as you step through the grib file.  Why?  The grib file interval is 3 hours, and the total time elapsed is 21 hours. This wxrte configuration time interval is set for 10 minute intervals, and weather_routing interpolates the grib file while calculating the route. 

Then when I try to step through the route using the grib_pi controller for time, the interval I must use is set by the grib files' 3 hours interval.  That results in three one locations where the boat position "round circle" shows on the route because the total time is 21 hours.

I wonder if there is a need to use the interpolated 10 minute intervals?  Because of the large difference betweeen 10 minutes and 3 hours.  

In all these years, I have not often done a 10 minute interval. It is more often a wxrte 1 hr interval against a 3hr grib file, or wxrte 3hr interval against a 6hr grib file.  In those cases I don't really mind the skipped positions on the route. I still get a good sense of the condtions.  

This particular case is a little different  (10min interval against a 3hr grib file).  This is not going to be a frequent use, I believe, but it does illustrate nicely the wxrte interval against the grib interval.

**Comment by stelian42** [comment]
From your screencap I see the arrival time is 21:33, for a departure at 12:00. This means a duration of 9 hours, or three intervals of 3 hours. Consistent with the 3 steps in grib controller.

If you try with the default configuration (look at my screencaps) and you only change from climatology to grib, and set the time to the start time of grib, you'll have a nice route and you'll be able to do many steps in the grib controller....

**Comment by rgleason** [comment]
Stelian. Yes, thanks for the correction. 

If you try with the default configuration (look at my screencaps) and you only change from climatology to grib, and set the time to the start time of grib, you'll have a nice route and you'll be able to do many steps in the grib controller....

I've downloaded your grib, and wanted to get the PI going first. Will try your example.

I've added this to the wiki FAQ 
https://opencpn.org/wiki/dokuwiki/doku.php?id=opencpn:opencpn_user_manual:plugins:weather:weather_routing#weather_routing_time_intervals_and_grib_data_time_intervals

I don't know if more explanation is needed.

I think there is a better place for this useful information, ...but this document needs some new screenshots due to changes in configuration menu and needs to be simplified.  The FAQ and Irma examples need to become separate sub pages and our new Climatology configuration needs to be illustrated, etc.

**Comment by rgleason** [comment]
At some point, I think we need to use one of the Climatology examples with a screenshot and explain this more completely.

> From your screencap I see the arrival time is 21:33, for a departure at 12:00. This means a duration of 9 hours, or three intervals of 3 hours. Consistent with the 3 steps in grib controller.

**Comment by stelian42** [comment]
I'm not sure I understand your explanation. For me it's simpler than that: when you step through the grib you will see the boat position at that current grib time (not "the multipier of the two time intervals"). This can be right on an isochrone or between two isochrones (if grib interval is smaller than wxrte time interval).

**Comment by rgleason** [comment]
Yes, I think that is a reasonable way to explain it too, and perhaps more accurate isofar as all cases.

I've used the big 33gb world grib and done one routing. Your illustration of multiple routings all at once is nice.  

What I realized is by stepping through the grib file, of course we can go no further than the extent of the grib file. 

![screenshot 356](https://user-images.githubusercontent.com/2363959/35285170-ad10ff42-002a-11e8-8d8e-2b980f3f305b.png)

We cannot change the date/time to step through the Climatology file, even when I turn on Climatology "wind"  

![screenshot 357](https://user-images.githubusercontent.com/2363959/35285178-b55b67d2-002a-11e8-8cf5-86bf43c2e430.png)


There are some nice squares showing sail changes. How do we find out more about those?

To bad we couldn't have a more universal Time Stepper Control.
